### PR TITLE
Handle nested messages and nested enums without repetitions

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1736
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1735

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

**Review by lahirusamith — COMMENTED** [review]
LGTM!

**Review by dilanSachi — APPROVED** [review]
LGTM

File: native/src/main/java/io/ballerina/stdlib/grpc/builder/stub/Message.java [hunk 1]
```diff
@@ -156,8 +183,49 @@ public Message build() {
...
+            if (fieldType.toString().isBlank()) {
+                fieldType.append(types[i]);
+            } else {
+                fieldType.append("_").append(types[i]);
+            }
+        }
+        return fieldType.toString();
+    }
+
+    private static boolean isProtoType(String type) {
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/grpc/builder/stub/Message.java:209]:
```suggestion
    private static boolean isWellKnowProtoType(String type) {
```

File: native/src/main/java/io/ballerina/stdlib/grpc/builder/stub/Message.java [hunk 2]
```diff
@@ -122,14 +138,25 @@ public Message build() {
...
+                ).build();
                 enumList.add(enumMessage);
             }
 
             List<Field> fieldList = new ArrayList<>();
             Map<String, List<Field>> oneofFieldMap = new HashMap<>();
             for (DescriptorProtos.FieldDescriptorProto fieldDescriptorProto : messageDescriptor.getFieldList()) {
-                Field field = Field.newBuilder(fieldDescriptorProto).build();
+                Field field;
+                if (isProtoType((fieldDescriptorProto.getTypeName()))) {
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/grpc/builder/stub/Message.java:152]:
```suggestion
                if (isWellKnowProtoType((fieldDescriptorProto.getTypeName()))) {
```

**Review by daneshk — APPROVED** [review]
Add a minor comment. overall LGTM

File: native/src/main/java/io/ballerina/stdlib/grpc/builder/stub/Message.java (hunk 2 above)
Comment (MadhukaHarith92) [reply native/src/main/java/io/ballerina/stdlib/grpc/builder/stub/Message.java]: Updated in ad57369

File: native/src/main/java/io/ballerina/stdlib/grpc/builder/stub/Message.java (hunk 1 above)
Comment (MadhukaHarith92) [reply native/src/main/java/io/ballerina/stdlib/grpc/builder/stub/Message.java]: Updated in ad57369

**Review by daneshk — APPROVED** [review]

**Pull request merged**